src/frontends/tensorflow/src/op/slice.cpp

```diff
@@ -18,11 +18,24 @@ OutputVector translate_slice_op(const NodeContext& node) {
     auto start = node.get_input(1);
     auto size = node.get_input(2);
 
-    auto stop = make_shared<Add>(start, size);
+    // compute stop values in case non-negative sizes
+    auto stop_pos = make_shared<Add>(start, size);
 
-    auto one = make_shared<Constant>(element::i64, Shape{1}, 1);
-    auto shape = make_shared<ShapeOf>(start);
-    auto step = make_shared<Broadcast>(one, shape);
+    // compute stop values in case negative sizes
+    // since TensorFlow supports only -1 among negative sizes
+    // assign stop values to the data shape
+    auto input_shape = make_shared<ShapeOf>(input);
+    auto stop_neg = make_shared<ConvertLike>(input_shape, size);
+
+    // select the correct stop value based on a sign of size value
+    auto zeros = make_shared<Constant>(size.get_element_type(), Shape{}, 0);
+    auto negative_sizes_mask = make_shared<Less>(size, zeros);
+    auto stop = make_shared<Select>(negative_sizes_mask, stop_neg, stop_pos);
```

Comment:
can we add comment that we can try to use FloorMod operation here instead of Select subgraph to improve performance? But additional investigations are needed on benefits of this approach